### PR TITLE
b2cuserflows public preview extended

### PR DESCRIPTION
**Pull request by nickgmicrosoft**

### CODE REVIEW

File: api-reference/beta/api/b2cidentityuserflow-post-pagelayouts.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}
+```
+
+
+### Response
+>**Note:** The response object shown here might be shortened for readability.
+<!-- {
+  "blockType": "response",
+  "truncated": true,
+  "@odata.type": "Microsoft.Cpim.Api.DataModels.userFlowPageLayout"
```
Comment (hecthern) [inline api-reference/beta/api/b2cidentityuserflow-post-pagelayouts.md:88]: Please update the namespaces from "Microsoft.Cpim.DataModels" to "microsoft.graph", which is the public one

File: api-reference/beta/api/userflowpagelayout-availableversions.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
+---
+title: "userFlowPageLayout: availableVersions"
+description: "**TODO: Add Description**"
```
Comment (hecthern) [inline api-reference/beta/api/userflowpagelayout-availableversions.md:3]: For reference check portal documentation [Page layout](https://docs.microsoft.com/en-us/azure/active-directory-b2c/javascript-and-page-layout?pivots=b2c-user-flow)

This method is the equivalent of the options box show page layout [versions](https://docs.microsoft.com/en-us/azure/active-directory-b2c/page-layout) so the user can know the available and supported versions.

File: api-reference/beta/api/userflowpagelayout-availableversions.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  "name": "userflowpagelayoutthis.availableversions"
+}
+-->
+``` http
+GET https://graph.microsoft.com/beta/identity/b2cUserFlows/{b2cIdentityUserFlowId}/pageLayouts/{userFlowPageLayoutId}/availableVersions
+```
+
+
+### Response
+>**Note:** The response object shown here might be shortened for readability.
```
Comment (hecthern) [inline api-reference/beta/api/userflowpagelayout-availableversions.md:62]: Can we put an actual response and include in the comment that the version values will change for each page layout? 
i.e.:
`{
    "@odata.context": "https://graph.microsoft.com/beta/$metadata#Collection(Edm.String)",
    "value": [
        "1.2.2",
        "1.2.1",
        "1.2.0",
        "1.1.0",
        "1.0.0"
    ]
}`

File: api-reference/beta/api/userflowpagelayout-list.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+---
+title: "List userFlowPageLayouts"
```
Comment (hecthern) [inline api-reference/beta/api/userflowpagelayout-list.md:2]: Which is the difference between this one and "b2cidentityuserflow-list-pagelayouts.md" ?

File: api-reference/beta/api/userflowpagelayout-restoredefaultpageuri.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
+---
+title: "userFlowPageLayout: restoreDefaultPageUri"
+description: "**TODO: Add Description**"
```
Comment (hecthern) [inline api-reference/beta/api/userflowpagelayout-restoredefaultpageuri.md:3]: This is the equivalent of setting "Use custom page content" to "No" in Azure Portal 
[Page layout Doc](https://docs.microsoft.com/en-us/azure/active-directory-b2c/javascript-and-page-layout?pivots=b2c-user-flow)

File: api-reference/beta/api/userflowpagelayout-update.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+|Name|Description|
+|:---|:---|
+|Authorization|Bearer {token}. Required.|
+|Content-Type|application/json. Required.|
+
+## Request body
+[!INCLUDE [table-intro](../../includes/update-property-table-intro.md)]
+
+
+|Property|Type|Description|
```
Comment (hecthern) [inline api-reference/beta/api/userflowpagelayout-update.md:46]: Please check the design, we only allow to update "pageUri" and "version", all the other properties are read only
[userFlowPageLayout design](https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding?version=GBmaster&path=/reviews/2409-b2cuserflows/api.md&_a=preview&anchor=userflowpagelayout)

File: api-reference/beta/api/b2cidentityuserflow-post-pagelayouts.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
+---
+title: "Create userFlowPageLayout"
+description: "Create a new userFlowPageLayout object."
```
Comment (hecthern) [inline api-reference/beta/api/b2cidentityuserflow-post-pagelayouts.md:3]: Similar to Delete, we do not support POST action
check design doc [userflowpagelayout](https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding?version=GBmaster&path=/reviews/2409-b2cuserflows/api.md&_a=preview&anchor=userflowpagelayout)

File: api-reference/beta/resources/b2cidentityuserflow.md [hunk 8]
```diff
@@ -44,6 +44,8 @@ To help you set up the most common identity tasks for your applications, Azure A
 |[List identity providers](../api/b2cidentityuserflow-list-identityproviders.md) (deprecated)|[identityProvider](../resources/identityProvider.md) collection|Retrieve all identity providers in a B2C user flow.|
 |[Add identity provider](../api/b2cidentityuserflow-post-identityproviders.md) (deprecated)|None|Add an identity provider to a B2C user flow.|
 |[Delete identity provider](../api/b2cidentityuserflow-delete-identityproviders.md) (deprecated)|None|Remove an identity provider from a B2C user flow|
+|[List pageLayouts](../api/b2cidentityuserflow-list-pagelayouts.md)|[userFlowPageLayout](../resources/userflowpagelayout.md) collection|Get the userFlowPageLayout resources from the pageLayouts navigation property.|
+|[Create userFlowPageLayout](../api/b2cidentityuserflow-post-pagelayouts.md)|[userFlowPageLayout](../resources/userflowpagelayout.md)|Create a new userFlowPageLayout object.|
```
Comment (hecthern) [inline api-reference/beta/resources/b2cidentityuserflow.md:48]: Not supported 😄

File: api-reference/beta/resources/userflowpagelayout.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Contains details about the user flow's page layout.
+
+Azure Active Directory B2C (Azure AD B2C) periodically releases improvements and fixes with each new page layout version. Organizations can browse available AD B2C user flow page layouts and update them to the latest versions.
+
+> [!IMPORTANT]
+> We highly recommend you keep your page layout versions up-to-date so that all page elements reflect the latest security enhancements, accessibility standards, and your feedback.
+
+## Methods
```
Comment (hecthern) [inline api-reference/beta/resources/userflowpagelayout.md:23]: Use design as reference [userflowpagelayout](https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding?version=GBmaster&path=/reviews/2409-b2cuserflows/api.md&_a=preview&anchor=userflowpagelayout)

File: api-reference/beta/api/userflowpagelayout-delete.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
+---
+title: "Delete userFlowPageLayout"
```
[the remark this answers is not in the document]
Comment (AlexanderMars) [reply api-reference/beta/api/userflowpagelayout-delete.md:2]: [userflowpagelayout-delete.md](https://github.com/microsoftgraph/microsoft-graph-docs/pull/15820/files/2ea15118c873b88624f61ced4d42818720206654#diff-41feddc465c25e29dbaba0d27857c12720212b7710f39cc90b9cdbcebdcc0eb5) was auto generated by stub generator tool according to the schema. Removed.

File: api-reference/beta/api/userflowpagelayout-availableversions.md (hunk 3 above)
Comment (AlexanderMars) [reply api-reference/beta/api/userflowpagelayout-availableversions.md:62]: @hecthern, not sure about this one. @FaithOmbongi, what do you think?

Comment (FaithOmbongi) [reply api-reference/beta/api/userflowpagelayout-availableversions.md:62]: @hecthern, @AlexanderMars - Any useful additional context is always welcome for the customer. You can add this context in the description of the response body (as a line after line 61). Also, the response object (lines 73-77) should be an actual payload to replace the placeholder object - applies globally in this PR.

File: api-reference/beta/api/userflowpagelayout-update.md (hunk 6 above)
Comment (hecthern) [reply api-reference/beta/api/userflowpagelayout-update.md:46]: Can we include a mention of the read only properties? just to keep clear that user is not allowed to change them

File: api-reference/beta/resources/b2cidentityuserflow.md [hunk 11]
```diff
@@ -52,8 +52,10 @@ To help you set up the most common identity tasks for your applications, Azure A
 |id|String|The name of the user flow. This is a required value and is immutable after it's created. The name will be prefixed with the value of `B2C_1_` after creation.|
 |userFlowType|userFlowType|The [type of user flow](/azure/active-directory-b2c/user-flow-versions). The supported values for **userFlowType** are: `signUp`, `signIn`, `signUpOrSignIn`, `passwordReset`, `profileUpdate`, `resourceOwner`.|
 |userFlowTypeVersion|Single|The version of the user flow.|
+|isJavaScriptEnabled|Boolean|The property that determines whether JavaScript customization is enabled within the B2C user flow page layouts.|
 |isLanguageCustomizationEnabled|Boolean|The property that determines whether language customization is enabled within the B2C user flow. Language customization is not enabled by default for B2C user flows.|
 |defaultLanguageTag|String|Indicates the default language of the b2cIdentityUserFlow that is used when no `ui_locale` tag is specified in the request. This field is [RFC 5646](https://tools.ietf.org/html/rfc5646) compliant.|
+|tokenClaimsConfiguration|[userFlowTokenClaimsConfiguration](../resources/userflowtokenclaimsconfiguration.md)|**TODO: Add Description**|
```
Comment (hecthern) [inline api-reference/beta/resources/b2cidentityuserflow.md:58]: This is not part of pageLayouts but I think will be cover by Nick

File: api-reference/beta/resources/userflowpagelayout.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+|Method|Return type|Description|
+|:---|:---|:---|
+|[List userFlowPageLayouts](../api/userflowpagelayout-list.md)|[userFlowPageLayout](../resources/userflowpagelayout.md) collection|Get a list of the [userFlowPageLayout](../resources/userflowpagelayout.md) objects and their properties.|
+|[Get userFlowPageLayout](../api/userflowpagelayout-get.md)|[userFlowPageLayout](../resources/userflowpagelayout.md)|Read the properties and relationships of a [userFlowPageLayout](../resources/userflowpagelayout.md) object.|
+|[Update userFlowPageLayout](../api/userflowpagelayout-update.md)|[userFlowPageLayout](../resources/userflowpagelayout.md)|Update the properties of a [userFlowPageLayout](../resources/userflowpagelayout.md) object.|
+|[availableVersions](../api/userflowpagelayout-availableversions.md)|String collection|Show the supported version of page layout to provide admins and developers the exact values allowed to use.|
+|[restoreDefaultPageUri](../api/userflowpagelayout-restoredefaultpageuri.md)|None|Restore the default value of the "pageUri" property.|
+
+## Properties
+|Property|Type|Description|
```
Comment (hecthern) [inline api-reference/beta/resources/userflowpagelayout.md:33]: Can we include in the description which ones are read only?

File: api-reference/beta/api/userflowpagelayout-availableversions.md (hunk 3 above)
Comment (AlexanderMars) [reply api-reference/beta/api/userflowpagelayout-availableversions.md:62]: @hecthern, could you, please, share with me an actual payload? Thanks.

File: api-reference/beta/resources/userflowpagelayout.md [hunk 13]
```diff
@@ -0,0 +1,63 @@
+---
+title: "userFlowPageLayout resource type"
+description: The amount of page layouts can be modified, only the present set of entities allows updated for "pageUri" and "version"."
```
[the remark this answers is not in the document]
Comment (AlexanderMars) [reply api-reference/beta/resources/userflowpagelayout.md:3]: corrected

File: api-reference/beta/resources/userflowpagelayout.md (hunk 12 above)
Comment (AlexanderMars) [reply api-reference/beta/resources/userflowpagelayout.md:33]: done

File: api-reference/beta/api/identitycontainer-list-b2cuserflows.md [hunk 14]
```diff
@@ -124,15 +124,15 @@ Content-type: application/json
           "userFlowType": "signUp",
           "userFlowTypeVersion": 1,
           "isLanguageCustomizationEnabled": false,
-          "defaultLanguageTag": null
+          "defaultLanguageTag": null,
```
Comment (hecthern) [inline api-reference/beta/api/identitycontainer-list-b2cuserflows.md:127]: This tailing "," will show a JSON syntaxis error.

File: api-reference/beta/resources/userflowsinglesignonsessionconfiguration.md [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+
+Defines single sign-on (SSO) behavior for an Azure AD B2C user flow. SSO adds security and convenience when users sign in across applications.
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|isSessionLifetimeAbsolute|Boolean|Indicates how a session is extended by the session lifetime setting or the Keep me signed in (KMSI) setting. `true` Indicates that the user is forced to re-authenticate after the time period specified. `false`  indicates that the session is extended every time the user performs a cookie-based authentication (default).|
+|sessionLifetimeInMinutes|Int32|The amount of time the Azure AD B2C session cookie is stored on the user's browser after successful authentication. You can set the session lifetime up to 1440 minutes (24 hours).|
+|isKeepMeSignedInEnabled|Boolean|Represents whenever "keep me sign in" is enabled or not, when `false`, "keepMeSignedInDays" is not modifiable.|
+<!-- |keepAliveInDays|Int32|The amount of days to keep the session active for local accounts.| -->
```
Comment (hecthern) [inline api-reference/beta/resources/userflowsinglesignonsessionconfiguration.md:24]: Let's remove this property (keepAliveInDays)

**Pull request closed without merging**